### ISSUE TITLE
[New Pattern] Repetitive method calls

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1189]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | New Pattern                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.12 |
| Language level | PHP 7.3                        |

Current behaviour:
---

Here is some example code:

```php
<?php

declare(strict_types=1);

final class FooBar1
{
    public function getFooBar2(): FooBar2
    {
        return new FooBar2();
    }
}

final class FooBar2
{
    public function getFooBar3(): FooBar3
    {
        return new FooBar3();
    }

    public function getString2(): string
    {
        return 'string2';
    }
}

final class FooBar3
{
    public function getString1(): string
    {
        return 'string1';
    }

    public function getString3(): string
    {
        return 'string3';
    }
}

$fooBar1 = new FooBar1();

// 1. case
$differentNestingLevelsNotReported = [
    $fooBar1->getFooBar2()->getFooBar3()->getString1(),
    $fooBar1->getFooBar2()->getString2()
];

// 2. case
$repeatedUsageAsKeysNotReported = [
    $fooBar1->getFooBar2()->getFooBar3()->getString1() => 1,
    $fooBar1->getFooBar2()->getFooBar3()->getString3() => 1
];
```

No warning is displayed in both cases.

Suggested behaviour:
---

**1.) Case**

`$fooBar1->getFooBar2()` can also be replaced by a local variable in this case.

It works fine, if above code is changed to

```php
$reportedNow = [
    $fooBar1->getFooBar2()->getFooBar3()->getString1(),
    $fooBar1->getFooBar2()->getFooBar3()->getString3()
];
```

**2.) Case**

`$fooBar1->getFooBar2()->getFooBar3()` in the key usage should be reported as well and can be replaced by a local variable in this case.

**Comment by kalessil** [comment]
Implemented!
